src/main/java/org/folio/dew/batch/bulkedit/jobs/TenantResolver.java

```
@@ -0,0 +1,83 @@
+    var jobId = jobExecution.getJobParameters().getString(JobParameterNames.JOB_ID);
+    var fileName = FilenameUtils.getName(jobExecution.getJobParameters().getString(FILE_NAME));
+    var affiliatedAndPermittedTenants = new HashSet<String>();
+    for (var tenantId : tenantIds) {
+      if (!affiliatedTenants.contains(tenantId)) {
+        var user = userClient.getUserById(folioExecutionContext.getUserId().toString());
+        var errorMessage = format(getAffiliationErrorPlaceholder(entityType), user.getUsername(),
+          resolveIdentifier(identifierType), itemIdentifier.getItemId(), tenantId);
+        bulkEditProcessingErrorsService.saveErrorInCSV(jobId, itemIdentifier.getItemId(), errorMessage, fileName);
+      } else if (!isBulkEditReadPermissionExists(tenantId, entityType)) {
+        var user = userClient.getUserById(folioExecutionContext.getUserId().toString());
+        var errorMessage = format(getViewPermissionErrorPlaceholder(entityType), user.getUsername(),
+          resolveIdentifier(identifierType), itemIdentifier.getItemId(), tenantId);
+        bulkEditProcessingErrorsService.saveErrorInCSV(jobId, itemIdentifier.getItemId(), errorMessage, fileName);
+      } else {
+        affiliatedAndPermittedTenants.add(tenantId);
+      }
+    }
+    return affiliatedAndPermittedTenants;
+  }
+
+  private boolean isBulkEditReadPermissionExists(String tenantId, EntityType entityType) {
+    try {
+      return permissionsValidator.isBulkEditReadPermissionExists(tenantId, entityType);
+    } catch (FeignException e) {
+      throw new BulkEditException(e.getMessage());
+    }
+  }
+
+  protected String getAffiliationErrorPlaceholder(EntityType entityType) {
+    if (entityType == EntityType.ITEM) {
+      return NO_ITEM_AFFILIATION;
+    }
+    return NO_HOLDING_AFFILIATION;
+  }
+
+  protected String getViewPermissionErrorPlaceholder(EntityType entityType) {
+    if (entityType == EntityType.ITEM) {
```

To protect, please, throw unsupported exception for different to items and holdings cases.

---

Updated